tests/e2e-tests/src/orchestration.test.ts

```diff
@@ -100,12 +101,20 @@ describe('orchestration', () => {
     const result = await orchestrationToolCalling();
     expect(result.getFinishReason()).toBe('tool_calls');
 
-    const tool_calls =
-      result.data.orchestration_result.choices[0].message.tool_calls;
+    const assistantMessage = result.getAssistantMessage();
 
-    expect(tool_calls).toHaveLength(1);
-    expect(tool_calls[0].function.name).toBeDefined();
-    expect(tool_calls[0].function.arguments).toBeDefined();
+    if (assistantMessage.tool_calls) {
+      expect(assistantMessage.tool_calls).toHaveLength(1);
+      expect(assistantMessage.tool_calls[0].function.name).toBeDefined();
+      expect(assistantMessage.tool_calls![0].function.arguments).toBeDefined();
+    } else {
+      throw new Error('Expected tool_calls to be defined');
+    }
+  });
+
+  it('should complete a chat when message history with tool calls is passed', async () => {
+    const result = await orchestrationMessageHistoryWithToolCalling();
+    assertContent(result);
```

Comment:
[pp] In general, I would make e2e test more concise. This test case covers the previous chat completion test, and overlaps a bit with the tool call one. I would maybe move the tool call test into this one and remove the previous chat completion one. E2E should be considered both from a technical perspective (from client to service) and scenario perspective (a real process including tool calls, chat completion and message history).